app/summary/services/summaryStateService.js

```diff
@@ -349,6 +349,50 @@ summaryApp.service('summaryStateService', function ($rootScope, $log, Course, Sc
 							}
 						});
 
+						// Filling in hash data for supportAssignments and supportStaff
+						var supportAssignments = {
+							list: {},
+							ids: []
+						};
+
+						action.payload.supportAssignments.forEach(function(supportAssignment) {
+							supportAssignments.list[supportAssignment.id] = supportAssignment;
+							supportAssignments.ids.push(supportAssignment.id);
+						});
+
+						var supportStaffList = {
+							list: {},
+							ids: []
+						};
+
+						action.payload.supportStaffList.forEach(function(staff) {
+							supportStaffList.list[staff.id] = staff;
+							supportStaffList.ids.push(staff.id);
+						});
+
+						// Add TA data to sectionGroups
+						slotSectionGroup.teachingAssistants = [];
+
+						action.payload.supportAssignments.forEach(function(supportAssignment) {
+							if (supportAssignment.appointmentType == "teachingAssistant" && supportAssignment.sectionGroupId == sectionGroup.id) {
+
+								var supportStaff = supportStaffList.list[supportAssignment.supportStaffId];
+
+								slotSectionGroup.teachingAssistants.push({
```

Comment:
Re-do this forEach loop + if statement as: slotSectionGroup.teachingAssistants = action.payload.supportAssignments.filter(the_if_conditions).map(the_push_contents). Cleaner, right?

Reply:
I would agree, looks like a straightforward candidate for filter.